rpcs3/Emu/Cell/Modules/cellGame.cpp

```diff
@@ -230,18 +230,13 @@ disc_change_manager::~disc_change_manager()
 
 error_code disc_change_manager::register_callbacks(vm::ptr<CellGameDiscEjectCallback> func_eject, vm::ptr<CellGameDiscInsertCallback> func_insert)
 {
-	if (!func_eject || !func_insert)
-	{
-		return CELL_GAME_ERROR_PARAM;
-	}
-
 	std::lock_guard lock(mtx);
 
 	eject_callback = func_eject;
 	insert_callback = func_insert;
 
-	Emu.GetCallbacks().enable_disc_eject(true);
-	Emu.GetCallbacks().enable_disc_insert(false);
+	Emu.GetCallbacks().enable_disc_eject(!!func_eject);
+	Emu.GetCallbacks().enable_disc_insert(!!func_insert);
```

Comment:
This makes zero sense.

Reply:
Oh I see.

Reply:
I think the eject enabling had the same bug if there is no disc.

Reply:
The whole thing is only supposed to work if the game is a disc game.
So it is guaranteed that a disc is present when the callback is registered.
The game would've been terminated if it was ejected beforehand.
So it has to eject the disc before it can insert a new one

Reply:
The only way to enable this for HDD games would be to check here if bdvd is mounted and then enable one of them depending on the mounting state and the presence of the respective callback.

Reply:
In any case it is wrong to enable both eject and insert.

Reply:
Additionally, insert should not be enabled on an eject if the insert callback is missing and vice versa